maximsan/nodejs-aws-be
# Task 9

**Evaluation criteria**

- [x] - A working and correct express application should be in the bff-service folder.
- [x] - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service. The response from the bff-service should be the same as if the recipient service was called directly.
- [x] - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
How to test:
Get products list
Create new product
Get products list - result shouldn’t have new product
Wait more than 2 minutes
Get products list - result should have new product

product service - https://g7jj8pkjc3.execute-api.eu-west-1.amazonaws.com/dev/products
cart service through API Gateway proxy - https://w9wglhsyph.execute-api.eu-west-1.amazonaws.com/dev/api/profile/cart

bff-service api - http://maximsan-bff-develop.eu-west-1.elasticbeanstalk.com
cart api service call via bff - http://maximsan-bff-develop.eu-west-1.elasticbeanstalk.com/cart
product api service call via bff - http://maximsan-bff-develop.eu-west-1.elasticbeanstalk.com/products

post body example: 
    {
        "price": 2499,
        "title": "Les Paul Standard '60s",
        "description": "Les Paul Standard '60s",
        "count": 1
    }
